File: LeetCode/2073/python/lc2073.py
from typing import List

# Runtime complexity: O(n)
# Auxiliary space complexity: O(1)
# Subjective level: medium-
# Solved on: 2026-02-09
class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        ans = 0
        for i in range(len(tickets)):
            kFactor = tickets[k] - 1 if i > k else tickets[k]
            ans += min(tickets[i], kFactor)
        return ans

File: LeetCode/2073/cpp/lc2073.cpp
#include <cassert>
#include <vector>

class Solution {
public:
  int timeRequiredToBuy(std::vector<int>& tickets, int k) {
    int ans = 0;
    for (int i = 0; i < tickets.size(); ++i) {
      ans += std::min(tickets[i], (i > k) ? tickets[k] - 1 : tickets[k]);
    }
    return ans;
  }
};

int main() {
  Solution s;
  {
    std::vector<int> tickets = {2, 3, 2};
    int k = 2;
    assert(s.timeRequiredToBuy(tickets, k) == 6);
  }
  {
    std::vector<int> tickets = {5, 1, 1, 1};
    int k = 0;
    assert(s.timeRequiredToBuy(tickets, k) == 8);
  }
}
